mixer/adapter/kubernetesenv/cache.go

```diff
@@ -70,17 +84,73 @@ func newCacheController(clientset kubernetes.Interface, refreshDuration time.Dur
 			&v1.Pod{},
 			refreshDuration,
 			cache.Indexers{
-				"ip": ipIndex,
+				"ip": podIP,
+			},
+		),
+		appsv1RS: cache.NewSharedIndexInformer(
+			&cache.ListWatch{
+				ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
+					return clientset.AppsV1().ReplicaSets(namespace).List(opts)
+				},
+				WatchFunc: func(opts metav1.ListOptions) (watch.Interface, error) {
+					return clientset.AppsV1().ReplicaSets(namespace).Watch(opts)
+				},
+			},
+			&appsv1.ReplicaSet{},
+			refreshDuration,
+			cache.Indexers{},
+		),
+		appsv1beta2RS: cache.NewSharedIndexInformer(
+			&cache.ListWatch{
+				ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
+					return clientset.AppsV1beta2().ReplicaSets(namespace).List(opts)
```

Comment:
Shouldn't k8s auto convert v1beta2 to v1 for the same kind?

Reply:
does it?  If so, this indexer can be removed. I wasn't sure.

Reply:
I don't know the answer myself. @ayj might know. Can be fixed in a follow-up.